lib/index.ts

```diff
@@ -67,14 +68,22 @@ class PWMetrics {
 
     for (let runIndex of runs) {
       try {
-        metricsResults[runIndex] = await this.run();
+        const currentMetricResult: MetricsResults = await this.run();
+        currentMetricResult.error = this.resultHasExpectationErrors(currentMetricResult);
```

Comment:
Thanks for the feedback. I'm simplifying the method and adding some types for `this.expectations` because it changes between `ExpectationMetrics` and `NormalizedExpectationMetrics` in the constructor.